service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/RemoveManagedIdentitiesAzureVmStep.java

```diff
@@ -33,8 +34,17 @@ public StepResult deleteResource(FlightContext context) {
             .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
 
     ComputeManager computeManager = crlService.getComputeManager(azureCloudContext, azureConfig);
-    return AzureVmHelper.removeAllUserAssignedManagedIdentitiesFromVm(
-        azureCloudContext, computeManager, resource.getVmName());
+    var helperResult =
```

Comment:
I considered some other options, such as refactoring the helper to return something besides a `StepResult`, inspecting failed `StepResult`s in the error handling in the super-class, or adding a generic existence check for resources. 
They each had their merits, but would all require significantly more planning and verification than I was prepared to do as part of this work.